src/qt/rpcconsole.cpp

```diff
@@ -414,7 +415,33 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
     {
         std::string result;
         std::string executableCommand = command.toStdString() + "\n";
-
+        std::regex validRegex("generate\\s*[\\(]?\\s*(\\d+)?\\s*[,]?\\s*(\\d+)?\\s*[\\)]?\\s*");
+        std::regex invalidRegex("generate[\\s\\(]+.*");
+        std::smatch match;
+        if (std::regex_match(executableCommand, match, validRegex)) {
+            const std::string nblocks{match.str(1) == "" ? "1" : match.str(1)};
+            const std::string maxtries{match.str(2) == "" ? "1000000" : match.str(2)};
```

Comment:
Sorry, just realised, perhaps you don't need to set `maxtries `variable with the default value as in `mining.h`, `DEFAULT_MAX_TRIES`, it's already set if maxtries is empty in the `generatetoaddress` command at `mining.cpp`, so:

```suggestion
            const std::string maxtries = match.str[2];
```

Reply:
Thanks, I'll test it and perhaps update the code in the next commit.